# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpcom/base/nsMemoryReporterManager.cpp
# Commit: 60cccadbeaf0
# Full Hash: 60cccadbeaf01eee779400d8ae6f0c6e5f80da1c
# Author: Gabriele Svelto <gsvelto@mozilla.com>
# Date: 2019-11-01 16:13:11
# Regressor Bug: 1586236
# File Overlap Count: 1
# Description:
#   Bug 1586236 - Use memory resource notifications to detect low memory scenarios on Windows; r=dmajor
#   
#   This patch uses the low memory resource notification facility to detect
#   scenarios where physical memory is running low without polling. This is a
#   significant change compared to the previous behavior which measured both
# ==============================================================================

diff -r 23c113d65b48 -r 60cccadbeaf0 xpcom/base/nsMemoryReporterManager.cpp
--- a/xpcom/base/nsMemoryReporterManager.cpp	Fri Nov 01 06:03:35 2019 +0000
+++ b/xpcom/base/nsMemoryReporterManager.cpp	Thu Oct 31 13:55:42 2019 +0000
@@ -2491,16 +2491,6 @@
 }
 
 NS_IMETHODIMP
-nsMemoryReporterManager::GetLowMemoryEventsVirtual(int64_t* aAmount) {
-  return GetInfallibleAmount(mAmountFns.mLowMemoryEventsVirtual, aAmount);
-}
-
-NS_IMETHODIMP
-nsMemoryReporterManager::GetLowMemoryEventsCommitSpace(int64_t* aAmount) {
-  return GetInfallibleAmount(mAmountFns.mLowMemoryEventsCommitSpace, aAmount);
-}
-
-NS_IMETHODIMP
 nsMemoryReporterManager::GetLowMemoryEventsPhysical(int64_t* aAmount) {
   return GetInfallibleAmount(mAmountFns.mLowMemoryEventsPhysical, aAmount);
 }
@@ -2744,8 +2734,6 @@
 DEFINE_REGISTER_DISTINGUISHED_AMOUNT(Infallible, StorageSQLite)
 DEFINE_UNREGISTER_DISTINGUISHED_AMOUNT(StorageSQLite)
 
-DEFINE_REGISTER_DISTINGUISHED_AMOUNT(Infallible, LowMemoryEventsVirtual)
-DEFINE_REGISTER_DISTINGUISHED_AMOUNT(Infallible, LowMemoryEventsCommitSpace)
 DEFINE_REGISTER_DISTINGUISHED_AMOUNT(Infallible, LowMemoryEventsPhysical)
 
 DEFINE_REGISTER_DISTINGUISHED_AMOUNT(Infallible, GhostWindows)